pandas/core/reshape/pivot.py

```diff
@@ -612,12 +612,17 @@ def _normalize(table, normalize, margins, margins_name="All"):
 
     elif margins is True:
 
+        # keep index and column of pivoted table
+        table_index = table.index
+        table_columns = table.columns
+        # drop margins created in pivot_table and only keep the core
         column_margin = table.loc[:, margins_name].drop(margins_name)
-        index_margin = table.loc[margins_name, :].drop(margins_name)
+        # separate cases between multiindex and index
+        if isinstance(table_index, MultiIndex):
+            index_margin = table.loc[margins_name, :].drop(margins_name, axis=1)
```

Comment:
why are we not always passing axis= here? I don't like the need for this if/then

Reply:
Yeah, i figured out a way to walk it around, and also avoid MI problem, seems even test speed is faster ^^ @jreback